tests/test_run.py

```diff
@@ -42,3 +42,24 @@ def test_run_script_from_internet(pipx_temp_env, capsys):
             "6bdfbb6e9c1132b1c38fdd2f195d4a24c540c324/pipx-demo.py",
         ]
     )
+
+
+def test_appargs_doubledash(capsys):
+    run_pipx_cli(["run", "pycowsay", "--", "hello"])
+    captured = capsys.readouterr()
+    assert "< -- hello >" not in captured.out
+    run_pipx_cli(["run", "pycowsay", "hello", "--"])
+    captured = capsys.readouterr()
+    assert "< hello -- >" not in captured.out
+    run_pipx_cli(["run", "pycowsay", "--"])
+    captured = capsys.readouterr()
+    assert "< -- >" not in captured.out
+    run_pipx_cli(["run", "--", "pycowsay", "--", "hello"])
+    captured = capsys.readouterr()
+    assert "< -- hello >" not in captured.out
+    run_pipx_cli(["run", "--", "pycowsay", "hello", "--"])
+    captured = capsys.readouterr()
+    assert "< hello -- >" not in captured.out
+    run_pipx_cli(["run", "--", "pycowsay", "--"])
+    captured = capsys.readouterr()
+    assert "< -- >" not in captured.out
```

Comment:
I think in this case we would be better off with unit tests of `parse_args` rather than e2e tests. 

i.e. monkeypatch sys.argv and call `parse_args` with the various permutations you have here.